odoo/tools/mimetypes.py

```diff
@@ -109,6 +109,11 @@ def _check_svg(data):
     if b'<svg' in data and b'/svg>' in data:
         return 'image/svg+xml'
 
+def _check_webp(data):
+    """This checks the presence of the WEBP and VP8* tags in the RIFF"""
+    offset = data.find(b'WEBPVP8')
```

Comment:
That seems like a very expensive check if someone decides to e.g. store a large MIDI file as we're going to check throughput the entire thing for that string. *And* it's subject to false positives.

Also [pillow doesn't seem to care much about the possibility of ancillary chunks, if my reading is correct?](https://github.com/python-pillow/Pillow/blob/95cff6e959bb3c37848158ed2145d49d49806a31/src/PIL/WebPImagePlugin.py#L25-L26)

Reply:
[Reading the webp specs](https://developers.google.com/speed/webp/docs/riff_container#riff_file_format) it doesn't look like arbitrary ancillary chunks are a thing at all as far as webp is concerned: if there *are* ancillary chunks, their possible existence follows from the data of the `VP8X` chunk, so Pillow's check seems correct: the b`WEBP` and b`VP8` CCs appear at offsets 8 and 12 for a VP8 file, and that's it. There is no need for seeking throughout the entire file.

Reply:
Now checking at specific offsets.